sdk/src/azure/iot/az_iot_provisioning_client.c

```diff
@@ -276,6 +277,35 @@ AZ_INLINE az_result _az_iot_provisioning_client_parse_payload_error_code(
   return AZ_ERROR_ITEM_NOT_FOUND;
 }
 
+AZ_INLINE az_result
+_az_iot_provisioning_client_get_json_object_span(az_json_reader* jr, az_span* out_span)
+{
+  *out_span = jr->token.slice;
+
+  if (jr->token.kind == AZ_JSON_TOKEN_NULL)
+  {
+    *out_span = AZ_SPAN_EMPTY;
+    return AZ_OK;
+  }
+
+  if (jr->token.kind != AZ_JSON_TOKEN_BEGIN_OBJECT)
+  {
+    return AZ_ERROR_UNEXPECTED_CHAR;
+  }
+
+  _az_RETURN_IF_FAILED(az_json_reader_skip_children(jr));
+
+  if (jr->token.kind != AZ_JSON_TOKEN_END_OBJECT)
+  {
+    return AZ_ERROR_UNEXPECTED_CHAR;
+  }
```

Comment:
I don't believe this check is necessary, since `az_json_reader_skip_children` should validate that it reached "end of container" (at the same depth), or fail otherwise.

Unless you observed some test case which shows otherwise, I think this branch is unreachable and can be removed.

Reply:
Replacing with a precondition.